internal/workflow/processing.go

```diff
@@ -523,11 +527,33 @@ func (w *ProcessingWorkflow) SessionHandler(sessCtx temporalsdk_workflow.Context
 
 	// Delete local temporary files.
 	defer func() {
+		// We need disconnected context here because when session gets released the cleanup
+		// activities get scheduled and then immediately canceled.
+		cleanUpContext, cancel := temporalsdk_workflow.NewDisconnectedContext(sessCtx)
+		defer cancel()
 		if tinfo.Bundle.FullPathBeforeStrip != "" {
-			activityOpts := withActivityOptsForLocalAction(sessCtx)
-			_ = temporalsdk_workflow.ExecuteActivity(activityOpts, activities.CleanUpActivityName, &activities.CleanUpActivityParams{
+			activityOpts := withActivityOptsForLocalAction(cleanUpContext)
+			if err := temporalsdk_workflow.ExecuteActivity(activityOpts, activities.CleanUpActivityName, &activities.CleanUpActivityParams{
 				FullPath: tinfo.Bundle.FullPathBeforeStrip,
-			}).Get(activityOpts, nil)
+			}).Get(activityOpts, nil); err != nil {
+				w.logger.Error(err, "failed to clean up", "path", tinfo.Bundle.FullPathBeforeStrip)
+			}
+		}
+		if tempBlob != "" {
```

Comment:
Calling CleanUpActivity multiple times works, but there is some unnecessary overhead for temporal to schedule the activity and for the worker to pull it off the queue each time the activity runs.  A more efficient option would be to have the clean up activity clean up all the temp files/directories at the same time.  You could use https://github.com/artefactual-sdps/temporal-activities/blob/main/filesys/remove_activity.go or modify the local CleanUpActivity.

Reply:
Thanks for the feedback. Changes are here: 1c3f844afa7244bbabefe58d6cc6b6687086aec5